### ISSUE TITLE
Some questions

### ISSUE THREAD

**Issue by aiyadehaijiao** [JosephKJ/OWOD#28]
1.In the clustering stage, are unknown classes only divided into one class? if so, for different kinds of unknown classes(some of which may have different features), how to reduce the distance between unknown classes?
2.In the contrastive clustering loss function, are prototypes updated in the previous stage labels for the next stage?
3.The goal of contrastive clustering is to reduce the distance within class and increase the distance between classes.
![image](https://user-images.githubusercontent.com/35215541/113649837-be454f80-96c1-11eb-8894-e19cffded37d.png)

In the formula, otherwise that I understand is to calculate the distance between classes, but only 
calculate the loss that the distance between two classes is not large. But if this part is taken into account, it will shorten the distance between a feature in one category and another. How can this be explained?
4.The labels of unknowns are selected top k boxes according to score from the background classes generated by PRN. But what are unknown region proposals? Do they contain all background boxes?
5.In the algorithm COMPUTECLUSTERINGLOSS, instead of using all original dataset to train the whole network, it retains the previous parameters by assigning weights. It is the contribution of these parameters that enables the network to remember how to identify known classes. In my opinion, this kind of knowledge prolongs the memory time of the network. When more and more new classes are introduced, the network will eventually forget the known classes. Can I understand that?

Thank you!

**Comment by JosephKJ** [comment]
> unknown classes only divided into one class

Yes, thats how we do now. Having sub-clusters for unknowns might be a good direction to explore to improve ORE. 


> are prototypes updated in the previous stage labels for the next stage

We are not explicitly doing it, but implicitly feature store does it.



> How can this be explained?

Distance with the same class prototype will be reduced and the distance with the other class prototypes will be increased. Does this clarify?



> what are unknown region proposals

Region proposals with high objectness score, but with no GT with significant IoU qualify as an unknown region proposal. 



> Do they contain all background boxes?

No, top-k.



> It is the contribution of these parameters that enables the network to remember how to identify known classes. Can I understand that?

The contrastive clustering only aids the energy based unknown identifier, which actually does the unknown identification. Kindly refer to section 4.3 of the paper. 

Kindly re-open this issue for further discussion.
